vendor/bam/src/json/frame.rs

```diff
@@ -136,14 +116,34 @@ impl FrameHandler<json::Frame> for JsonFrameHandler {
 
                 Ok(None)
             }
-            _ => Err(Error::UnknownFrameType(frame._type)),
+            _ => Err(Error::UnknownFrameType(frame.frame_type)),
         }
     }
 }
 
+fn malformed_request(error: serde_json::Error) -> json::Response {
+    warn!("incoming request was malformed: {:?}", error);
+
+    json::Response::new(Status::SE(0))
+}
+
+fn unknown_request_type(request_type: &str) -> json::Response {
+    warn!("request type '{}' is unknown", request_type);
+
+    json::Response::new(Status::SE(2))
+}
+
+fn unknown_mandatory_headers(unknown_headers: UnknownMandatoryHeaders) -> json::Response {
```

Comment:
❓should we log this here as well?
```suggestion
fn unknown_mandatory_headers(unknown_headers: UnknownMandatoryHeaders) -> json::Response {
    warn!("incoming request had unknown mandatory headers '{}' ", unknown_headers);
```

Reply:
Since it is a well-defined error message that is returned to the client, debugging that should be pretty straight forward.